CMakeLists.txt

```diff
@@ -1142,6 +1142,14 @@ if(HPX_WITH_NETWORKING)
     CATEGORY "Build Targets"
     ADVANCED
   )
+  hpx_option(
+    HPX_WITH_FETCH_NANOBENCH
+    BOOL
+    "Use FetchContent to fetch Nanobench. (default: OFF)"
+    OFF
+    CATEGORY "Build Targets"
+    ADVANCED
+  )
```

Comment:
Should we fetch this by default only if `HPX_WITH_TESTS_BENCHMARKS=On`?

Reply:
Yes, so if hpx is built with benchmarks on, by default we fetch nanobench unless the user species the nanobench root directory. 

Should the default case be the non-nanobench version?